VirtoCommerce.Domain/Catalog/Model/CatalogProduct.cs

```diff
@@ -162,5 +163,62 @@ public virtual void ReduceDetails(string responseGroup)
                 Variations = null;
             }
         }
+
+        #region ICloneable members
+        public virtual object Clone()
+        {
+            var result = MemberwiseClone() as CatalogProduct;
+
+            if (Properties != null)
```

Comment:
Please use shorter syntax  in similar cases:
result.Properties = Properties?.Select(x => x.Clone() as Property) ?? result.Properties;